src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -209,7 +214,16 @@ private IEnumerable<IConfigurationSection> GetFeatureDefinitionSections()
             }
             else
             {
-                return _configuration.GetChildren();
+                if (_options.RequireFeatureManagementSection)
+                {
+                    _logger.LogWarning($"No configuration section named '{FeatureManagementSectionName}' was found.");
```

Comment:
If someone is setting that setting I would expect it is because they have a use case where they are running without feature flags at some point. I think a warning logged is just going to be noise.

Reply:
It's only logged when RequireFeatureManagementSection is set to true and there is no section. To me, it makes sense to warn because of the flag.

Reply:
I agree with Ross.
By default, "RequireFeatureManagementSection" flag is false. If this flas is set to true, it indicates that the user intentionally did so. In this case, since the users are using FeatureManagment lib, they should add "FeatureManagement" section in the configuration. If we cannot find this section, there must be something wrong. Maybe, the section name is misspelled, which is a unintentional error. So, I think the warning log is necessary. Besides, it also follows the behavior of other feature management option flags.